src/cargo/core/compiler/custom_build.rs

```diff
@@ -74,11 +75,83 @@ pub enum Severity {
 
 pub type LogMessage = (Severity, String);
 
+#[derive(Clone, Debug, Hash, PartialEq, Eq)]
+pub enum LibraryPath {
```

Comment:
This enum and variants may be worth some documentations about what they are and why we're doing this. And I think we can derive Ord here.

https://doc.rust-lang.org/stable/std/cmp/trait.Ord.html#derivable

> When derived on enums, variants are ordered primarily by their discriminants. Secondarily, they are ordered by their fields. By default, the discriminant is smallest for variants at the top, and largest for variants at the bottom. Here’s an example: